src/gui/PasswordGeneratorWidget.cpp

```diff
@@ -251,44 +303,71 @@ PasswordGenerator::GeneratorFlags PasswordGeneratorWidget::generatorFlags()
-    if (flags.testFlag(PasswordGenerator::CharFromEveryGroup)) {
-        if (classes.testFlag(PasswordGenerator::LowerLetters)) {
-            minLength++;
+    if (m_ui->tabWidget->currentIndex() == Password) {
+        PasswordGenerator::CharClasses classes = charClasses();
+        PasswordGenerator::GeneratorFlags flags = generatorFlags();
+
+        int minLength = 0;
+        if (flags.testFlag(PasswordGenerator::CharFromEveryGroup)) {
+            if (classes.testFlag(PasswordGenerator::LowerLetters)) {
+                minLength++;
+            }
+            if (classes.testFlag(PasswordGenerator::UpperLetters)) {
+                minLength++;
+            }
+            if (classes.testFlag(PasswordGenerator::Numbers)) {
+                minLength++;
+            }
+            if (classes.testFlag(PasswordGenerator::SpecialCharacters)) {
+                minLength++;
+            }
         }
-        if (classes.testFlag(PasswordGenerator::UpperLetters)) {
-            minLength++;
-        }
-        if (classes.testFlag(PasswordGenerator::Numbers)) {
-            minLength++;
+        minLength = qMax(minLength, 1);
+
+        if (m_ui->spinBoxLength->value() < minLength) {
+            m_updatingSpinBox = true;
+            m_ui->spinBoxLength->setValue(minLength);
+            m_ui->sliderLength->setValue(minLength);
+            m_updatingSpinBox = false;
         }
-        if (classes.testFlag(PasswordGenerator::SpecialCharacters)) {
-            minLength++;
+
+        m_ui->spinBoxLength->setMinimum(minLength);
+        m_ui->sliderLength->setMinimum(minLength);
+
+        m_passwordGenerator->setLength(m_ui->spinBoxLength->value());
+        m_passwordGenerator->setCharClasses(classes);
+        m_passwordGenerator->setFlags(flags);
+
+        if (m_passwordGenerator->isValid()) {
+            m_ui->buttonGenerate->setEnabled(true);
+        } else {
+            m_ui->buttonGenerate->setEnabled(false);
         }
-    }
-    minLength = qMax(minLength, 1);
+    } else {
+        int minWordCount = 0;
 
-    if (m_ui->spinBoxLength->value() < minLength) {
-        m_updatingSpinBox = true;
-        m_ui->spinBoxLength->setValue(minLength);
-        m_ui->sliderLength->setValue(minLength);
-        m_updatingSpinBox = false;
-    }
+        if (m_ui->spinBoxWordCount->value() < minWordCount) {
+            m_updatingSpinBox = true;
+            m_ui->spinBoxWordCount->setValue(minWordCount);
+            m_ui->sliderWordCount->setValue(minWordCount);
+            m_updatingSpinBox = false;
+        }
 
-    m_ui->spinBoxLength->setMinimum(minLength);
-    m_ui->sliderLength->setMinimum(minLength);
+        m_ui->spinBoxWordCount->setMinimum(minWordCount);
+        m_ui->sliderWordCount->setMinimum(minWordCount);
 
-    m_generator->setLength(m_ui->spinBoxLength->value());
-    m_generator->setCharClasses(classes);
-    m_generator->setFlags(flags);
+        m_dicewareGenerator->setLength(m_ui->spinBoxWordCount->value());
```

Comment:
@TheZ3ro `setLength` is a bit ambiguous, since it's not really the length of the resulting passphrase, but instead its word count. In fact, the name of the spin box is wordCount.